fix(sbb-stepper): marker not visible in stepper

DavideMininni-Fincons opened the pull request:
## Preflight Checklist

<!-- Please ensure you've completed the following steps by replacing [ ] with [x]-->

- [x] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/sbb-design-systems/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #3401 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../docs/REVIEW.md) for more information on what is checked during the review process.

## Changes

Changes in this pull request:
if the sbb-stepper is used in a sbb-dialog, the marker on the selected element will not appear, because the calculations are done when the dialog is closed, so the marker has a width of 0.
We need to recalculate it when the element becomes visible and to do so a InterceptorController has been added.

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. -->

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

jeripeierSBB commented on src/elements/stepper/stepper/stepper.ts:
```diff
@@ -1,3 +1,4 @@
+import { IntersectionController } from '@lit-labs/observers/intersection-controller.js';
```
Only a simple question, would it work when adding setmarkersize to _onStepperResize ?

jeripeierSBB replied:
I mean instead of the intersectioncontroller

TomMenga replied:
I don't think it would be enough since the `window:resize` is not triggered when the dialog is opened

DavideMininni-Fincons replied:
I confirm

> I don't think it would be enough since the `window:resize` is not triggered when the dialog is opened

jeripeierSBB commented on src/elements/stepper/stepper/stepper.ts:
```diff
@@ -29,6 +30,22 @@ export
 class SbbStepperElement extends SbbHydrationMixin(LitElement) {
   public static override styles: CSSResultGroup = style;
 
+  /**
```
Could you create a test case covering the fix?

DavideMininni-Fincons replied:
yes, sure, i'll update the pr soonish

TomMenga reviewed (approved):
LGTM 👍 
As Jeri suggested, a visual spec of dialog + stepper would be nice

DavideMininni-Fincons commented:
PR updated with test
@jeripeierSBB @TomMenga 

Please note 
on the current (main) implementation there is this warning in the browser's console:
`step.ts:66 Element sbb-step scheduled an update (generally because a property was set) after an update completed, causing a new update to be scheduled. This is inefficient and should be avoided unless the next update can only be scheduled as a side effect of the previous update. See https://lit.dev/msg/change-in-update for more information.`

When I added the test with dialog + stepper, I received the mentioned warning and this error, which cause the test to fail:
`Error: ResizeObserver loop completed with undelivered notifications. (:0)`

This appears to be related to the sbb-step setting the height of the sbb-stepper via a ResizeObserver in the `firstUpdated` hook. To fix the error i removed the observer logic and I added a similar logic directly in the sbb-stepper: the warning is disappeared and the test does not fail anymore.

jeripeierSBB commented on src/elements/stepper/stepper/stepper.ts:
```diff
@@ -29,6 +30,24 @@ export
+    target: null,
+    callback: (entries) => {
+      entries.forEach((e) => {
+        if (e.intersectionRatio === 1) {
```
would that also work, if the element is half visible? or should we use something like "> 0"?

DavideMininni-Fincons replied:
In this specific case, i see the interceptor going directly to 1 when the dialog is open; i feel we can relax the check with > 0

jeripeierSBB reviewed (approved):
LGTM :) some minor stuff found

DavideMininni-Fincons commented on src/elements/dialog/dialog/dialog.spec.ts:
```diff
@@ -587,4 +589,64 @@ describe('sbb-dialog', () => {
+              .replaceAll('px', ''),
+          ) > 0,
+      );
+      console.log(getComputedStyle(stepper).getPropertyValue('--sbb-stepper-marker-size'));
```
whoops

DavideMininni-Fincons commented on src/elements/stepper/stepper/stepper.ts:
```diff
@@ -183,6 +203,21 @@ class SbbStepperElement extends SbbHydrationMixin(LitElement) {
+      if (innerElement) {
+        this.style?.setProperty(
+          '--sbb-stepper-content-height',
+          getComputedStyle(innerElement).getPropertyValue('height'),
```
No special thoughts behind getComputedStyle, I just remembered I used in some other test. Replaced.
